lib/hermetic/youtrack.ex

```diff
@@ -21,34 +21,92 @@ defmodule Hermetic.YouTrack do
   end
 
   @doc """
-    Return URL to YouTrack avatar for the given username.
+  Send HTTP PUT request to provided YouTrack endpoint.
+  """
+  def put!(endpoint) do
+    headers = [
+      {"authorization", "Bearer " <> token()},
+    ]
+
+    HTTPoison.put!(base_url() <> endpoint, "", headers)
+  end
+
+  @doc """
+  Send HTTP POST request to provided YouTrack endpoint.
+  """
+  def post!(endpoint) do
+    headers = [
+      {"authorization", "Bearer " <> token()},
+    ]
+
+    HTTPoison.post!(base_url() <> endpoint, "", headers)
+  end
+
+  @doc """
+  Create a new issue and return the issue id
+  """
+  @spec create_issue(String.t(), String.t(), String.t()) :: String.t()
+  def create_issue(project, summary, description) do
+    resp = put!("/rest/issue?" <> URI.encode_query([
+      project: String.upcase(project),
+      summary: summary,
+      description: description,
+    ]))
+    headers = Map.new(resp.headers)
```

Comment:
Use `Keyword.fetch!/2` instead.

Reply:
Seems like that doesn't work on string keys, which is what you get from httpoison

Reply:
You're right, I've marked this as resolved.